src/Controller/Api/Legacy/MessageController.php

```diff
@@ -102,19 +103,19 @@ public function getMessagesAction(ParamFetcherInterface $paramFetcher)
      *
      * @param Request $request A Symfony request
      *
-     * @return MessageInterface
+     * @return FormInterface
      */
     public function postMessageAction(Request $request)
     {
         $message = null;
 
-        $form = $this->formFactory->createNamed(null, 'sonata_notification_api_form_message', $message, [
+        $form = $this->formFactory->createNamed('', MessageSerializationType::class, $message, [
             'csrf_protection' => false,
         ]);
 
         $form->handleRequest($request);
 
-        if ($form->isValid()) {
+        if ($form->isSubmitted() && $form->isValid()) {
```

Comment:
I think this wasn't working, passing a form alias to `createNamed` was deprecated some versions ago and I think it was also mandatory in Symfony `4.0` to call `isSubmitted` before `isValid`.